python/hsfs/core/online_ingestion.py

```diff
@@ -0,0 +1,158 @@
+#
+#   Copyright 2024 Hopsworks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+from __future__ import annotations
+
+import json
+import time
+import warnings
+from datetime import datetime, timedelta
+from typing import Any, Dict, Optional
+
+import humps
+from hopsworks_common import util
+from tqdm.auto import tqdm
+
+
+class OnlineIngestion:
+    """
+    Metadata object used to provide Online Ingestion information for a feature group.
+    """
+
+    def __init__(
+        self,
+        id: Optional[int] = None,
+        num_entries: int = None,
+        current_offsets: int = None,
+        processed_entries: int = None,
+        inserted_entries: int = None,
+        aborted_entries: int = None,
+        batch_results = None,
+        **kwargs,
+    ):
+        self._id = id
+        self._num_entries = num_entries # specified when inserting
+        self._current_offsets = current_offsets
+        self._processed_entries = processed_entries
+        self._inserted_entries = inserted_entries
+        self._aborted_entries = aborted_entries
+        self._batch_results = batch_results # batch inserts performed by onlinefs
+
+    @classmethod
+    def from_response_json(cls, json_dict: Dict[str, Any]) -> "OnlineIngestion":
```

Comment:
Since __future__.annotations are imported, there is no need for the quotes around OnlineIngestion.